modules/apps/roles/roles-admin-impl/src/main/java/com/liferay/roles/admin/internal/security/permission/contributor/SiteRoleContributor.java

```diff
@@ -51,6 +61,52 @@ public void contribute(RoleCollection roleCollection) {
 		}
 	}
 
+	private boolean _userIsMemberOfAtLeastOneDepot(
+			Group group, RoleCollection roleCollection)
+		throws PortalException {
+
+		User user = roleCollection.getUser();
+
+		if (group.getCompanyId() != user.getCompanyId()) {
+			return false;
+		}
+
+		for (Group userGroup : user.getGroups()) {
+			if (userGroup.isDepot()) {
+				return true;
+			}
+		}
```

Comment:
You can replace this with `if (ListUtil.exists(user.getGroups(), UserGroup::isDepot) { return true }`.